src/data/nat/basic.lean

```diff
@@ -1024,6 +1057,75 @@ dvd_trans this hdiv
 lemma dvd_of_pow_dvd {p k m : ℕ} (hk : 1 ≤ k) (hpk : p^k ∣ m) : p ∣ m :=
 by rw ←nat.pow_one p; exact pow_dvd_of_le_of_pow_dvd hk hpk
 
+lemma eq_of_dvd_quot_one {a b : ℕ} (w : a ∣ b) (h : b / a = 1) : a = b :=
+begin
+  cases w with b w,
+  subst w,
+  rw nat.mul_comm at h,
+  rw nat.mul_div_cancel at h,
+  { subst h, simp },
+  { by_contradiction w, simp at w, subst w, simpa using h }
+end
+
+lemma div_succ_le_div {a b : ℕ} (h : b > 0) : a / (b + 1) ≤ a / b :=
```

Comment:
I don't think we need any of these except for `div_le_div_left`, and that should have an easy proof using the galois connection.

Reply:
Sorry, I don't know what to do with this one.

Reply:
@digama0, no hurry, but could you give me a hint for this?

Reply:
```lean
lemma div_le_div_left {a b c : ℕ} (h₁ : c ≤ b) (h₂ : 0 < c) : a / b ≤ a / c :=
(nat.le_div_iff_mul_le _ _ h₂).2 $
  le_trans (mul_le_mul_left _ h₁) (div_mul_le_self _ _)
```
also don't use `ge` / `gt`

Reply:
Thanks! I’ve incorporated this.